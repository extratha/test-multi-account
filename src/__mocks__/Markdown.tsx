const MockMarkdown = (props: any) => {
  return <div {...props} />;
};

jest.mock("react-markdown", () => MockMarkdown);
