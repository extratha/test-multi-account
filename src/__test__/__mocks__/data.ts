export const data = {
  test: 'test'
}