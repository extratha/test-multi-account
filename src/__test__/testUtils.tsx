import { RenderOptions, RenderResult, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { SWRConfig } from 'swr';
import { theme } from '@/config/config-mui/theme';
import { ThemeProvider as MUIThemeProvider } from '@mui/material/styles';
import { NextIntlClientProvider } from 'next-intl';
import commonTh from '../../public/locales/th/common.json';
import commonEn from '../../public/locales/th/common.json';
type RenderWithProvider = (
  elm: React.ReactElement,
  renderOptions?: ProviderOptions,
) => RenderResult;

export type ProviderOptions = {
  formOptions?: Record<string, unknown>;
} & RenderOptions;

const messages = {
  ...commonTh,
  ...commonEn,
};

const renderWithProviders: RenderWithProvider = (
  component,
  renderOptions?,
  locale = 'th',
) => {
  const wrapper = ({ children }: { children: React.ReactNode }) => (
    <NextIntlClientProvider locale={locale} messages={messages}>
      <MUIThemeProvider theme={theme}>
        <SWRConfig value={{ provider: () => new Map() }}>{children}</SWRConfig>
      </MUIThemeProvider>
    </NextIntlClientProvider>
  );

  return render(component, {
    wrapper,
    ...renderOptions,
  });
};

export { renderWithProviders as render, userEvent };
export * from '@testing-library/react';
