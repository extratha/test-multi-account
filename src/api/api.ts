import { ExampleDataResult, InterpretResult, SubmitLabInterpretsResponse } from "@/types/aiInterpret";
import { SubmitLabInterpretsRequest } from "@/types/interpretInputDataConfig";
import axiosInstance from "@/utils/axios";

export const apiAxios = axiosInstance;

export const getLabExampleList = () => {
  return apiAxios.get<ExampleDataResult[]>("/lab/examples");
};

export const getLabExampleId = (id: string) => {
  return apiAxios.get<InterpretResult>(`/lab/examples/${id}`);
};

export const submitLabInterprets = (data: SubmitLabInterpretsRequest) => {
  return apiAxios.post<SubmitLabInterpretsResponse>("/lab/interprets", data);
};

export const getLabInterpretsByTransactionId = (transactionId: string) => {
  return apiAxios.get<InterpretResult>(`/lab/interprets/${transactionId}`);
};
