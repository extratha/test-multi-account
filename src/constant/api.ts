
export const API = {
  PATH: {
    login: 'auth/login',
    setNewPassword: 'auth/reset-password',
    changePassword: 'auth/change-password',
    forgotPassword: 'auth/forgot-password',
    exampleDataList: 'lab/examples',
  },
  HTTP_METHODS: {
    GET: 'GET',
    POST: 'POST',
    PUT: 'PUT',
    PATCH: 'PATCH',
    DELETE: 'DELETE',
  },
  ERROR: {
    UNAUTHENTICATION: {
      MESSAGE: 'unauthenticated',
      CODE: 401,
    },
    login: {
      INVALID_CREDENTIAL: {
        MESSAGE: 'invalid email or password',
        CODE: 400,
      },
    }
  },
};
