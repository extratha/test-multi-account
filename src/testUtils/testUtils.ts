import { act, render as testRender } from "@testing-library/react";
import dayjs from "dayjs";
import flushPromises from "flush-promises";
import { ReactElement } from "react";

import { TestProvider } from "./TestProvider";

export const API = {
  LOGIN: "/auth/login",
  EXAMPLE_AI_INTERPRET_URL: "/lab/examples/labId",
  GET_LAB_INTERPRETS_ID_URL: "/lab/interprets/labId",
  SUBMIT_LAB_INTERPRETS_URL: "/lab/interprets",
  CONSENT: "/consents",
  TERMS_AND_CONDITIONS: "/consents/terms-conditions",
  TERMS_AND_CONDITIONS_LATEST: "/consents/terms-conditions/latest",
  PRIVACY_POLICY: "/consents/privacy-policies",
  PRIVACY_POLICY_LATEST: "/consents/privacy-policies/latest",
  PUBLIC_TERMS_CONDITIONS: "/consents/terms-conditions/public",
  PUBLIC_PRIVACY_POLICIES: "/consents/privacy-policies/public",
};

export const render = (ui: ReactElement) => {
  return testRender(ui, { wrapper: TestProvider });
};

export const flushPromise = async () => {
  await act(async () => {
    await flushPromises();
  });
};

export const advanceTimersByTime = async (time: number) => {
  await act(async () => {
    jest.advanceTimersByTime(time);
  });
};

export const setMockDate = (date: string) => {
  jest.setSystemTime(dayjs(date).valueOf());
};
