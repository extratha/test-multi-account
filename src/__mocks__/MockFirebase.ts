import { mockInputFieldConfig } from "./data";

jest.mock("firebase/app", () => ({
  initializeApp: jest.fn(),
}));

jest.mock("firebase/remote-config", () => ({
  getRemoteConfig: jest.fn().mockReturnValue({ settings: {} }),
  fetchAndActivate: jest.fn().mockResolvedValue(""),
  getValue: jest.fn().mockReturnValue({
    asString: () => mockInputFieldConfig,
  }),
}));
