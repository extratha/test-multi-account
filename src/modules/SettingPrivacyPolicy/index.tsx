"use client";

import { Box, Container, Divider, Paper, Typography, styled } from "@mui/material";
import { useEffect, useState } from "react";

import { getPrivacyPolicyLatest } from "@/api/api";
import ConsentContent from "@/components/ConsentContent";
import useTranslation from "@/locales/useLocale";
import { ConsentResultLatest } from "@/types/model.api";

const Wrapper = styled(Container)({
  position: "relative",
  flex: 1,
  maxWidth: "1024px",
});

// TODO: refactor layout
const WrapperLayout = styled(Box)({
  position: "absolute",
  left: 0,
  top: 0,
  width: "100%",
  height: "100%",
  padding: "24px",
});

const Content = styled(Paper)({
  width: "100%",
  height: "100%",
  padding: "36px 40px",
  borderRadius: "16px",
  boxShadow: "none",
  overflowY: "auto",
});

const TitleDivider = styled(Divider)({
  margin: "8px 0px 24px",
});

const SettingPrivacyPolicy = () => {
  const { translation } = useTranslation()

  const [consent, setConsent] = useState<ConsentResultLatest>();

  const fetchPrivacyPolicy = async () => {
    try {
      const { data } = await getPrivacyPolicyLatest();
      setConsent(data);
    } catch (error) {
      //TODO : handle error
    }
  };

  useEffect(() => {
    fetchPrivacyPolicy();
  }, []);

  return (
    <Wrapper>
      <WrapperLayout>
        {consent && (
          <Content data-testid="privacy-policy-consent">
            <Typography variant="titleLargeSemiBold">{translation("Common.settingPrivacyPolicy.title")}</Typography>
            <Typography variant="titleLargeSemiBold">{translation("Common.button.next")}</Typography>
            <TitleDivider />
            <ConsentContent name="privacy-policy" data={consent.consent} />
          </Content>
        )}
      </WrapperLayout>
    </Wrapper>
  );
};

export default SettingPrivacyPolicy;
