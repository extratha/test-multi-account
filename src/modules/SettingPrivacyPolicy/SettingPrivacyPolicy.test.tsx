import MockAdapter from "axios-mock-adapter";

import { mockPrivacyPolicyDataLatest } from "@/__mocks__/data";
import { apiAxios } from "@/api/api";
import { API, flushPromise, render } from "@/testUtils/testUtils";
import SettingPrivacyPolicy from ".";

describe("SettingPrivacyPolicy", () => {
  let mockApiAdapter: MockAdapter;

  beforeEach(() => {
    mockApiAdapter = new MockAdapter(apiAxios);
    mockApiAdapter.onGet(API.PRIVACY_POLICY_LATEST).reply(200, mockPrivacyPolicyDataLatest);
  });

  const renderPrivacyPolicy = async () => {
    const view = render(<SettingPrivacyPolicy />);
    await flushPromise();
    return view;
  };

  it("should render correctly", async () => {
    const { asFragment } = await renderPrivacyPolicy();
    expect(asFragment()).toMatchSnapshot();
  });
});
