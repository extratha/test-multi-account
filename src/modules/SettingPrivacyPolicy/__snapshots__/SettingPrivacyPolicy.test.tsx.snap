// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingPrivacyPolicy should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 100%;
  margin-left: auto;
  box-sizing: border-box;
  margin-right: auto;
  display: block;
  padding-left: 16px;
  padding-right: 16px;
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  max-width: 1024px;
}

@media (min-width:600px) {
  .emotion-0 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:1240px) {
  .emotion-0 {
    max-width: 1240px;
  }
}

.emotion-1 {
  position: absolute;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  padding: 24px;
}

.emotion-2 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 8px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  width: 100%;
  height: 100%;
  padding: 36px 40px;
  border-radius: 16px;
  box-shadow: none;
  overflow-y: auto;
}

.emotion-3 {
  margin: 0;
  font-weight: 500;
  line-height: 28px;
  letter-spacing: 0;
  font-size: 20px;
  font-family: NotoSansThai-SemiBold,sans-serif;
}

.emotion-4 {
  margin: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  border-width: 0;
  border-style: solid;
  border-color: rgba(0, 0, 0, 0.12);
  border-bottom-width: thin;
  margin: 8px 0px 24px;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-5>:not(style):not(style) {
  margin: 0;
}

.emotion-5>:not(style)~:not(style) {
  margin-top: 24px;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 24px;
  border-radius: 12px;
  background: #FAFAFA;
}

.emotion-7 {
  font-size: 16px;
  white-space: normal;
  color: #212121;
}

.emotion-7>*:first-of-type {
  margin-top: 0px;
}

.emotion-7>*:last-of-type {
  margin-bottom: 0px;
}

.emotion-9 {
  margin: 0;
  font-weight: 500;
  line-height: 28px;
  letter-spacing: 0;
  font-size: 20px;
  font-family: NotoSansThai-SemiBold,sans-serif;
  margin-bottom: 24px;
  color: #212121;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  border: 1px solid #C2C7CF;
  background-color: #fff;
}

.emotion-10:first-of-type {
  border-radius: 12px 12px 0 0;
}

.emotion-10:last-of-type {
  border-radius: 0 0 12px 12px;
}

.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  cursor: pointer;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 24px;
}

.emotion-13 {
  max-width: 52px;
  max-height: 52px;
}

.emotion-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-left: 12px;
}

.emotion-14>:not(style):not(style) {
  margin: 0;
}

.emotion-14>:not(style)~:not(style) {
  margin-top: 4px;
}

.emotion-15 {
  margin: 0;
  font-weight: 500;
  line-height: 28px;
  letter-spacing: 0;
  font-size: 20px;
  font-family: NotoSansThai-SemiBold,sans-serif;
  color: #212121;
}

.emotion-16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-17 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  padding: 8px;
  border-radius: 50%;
  overflow: visible;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-17::-moz-focus-inner {
  border-style: none;
}

.emotion-17.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-17 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-17:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-17:hover {
    background-color: transparent;
  }
}

.emotion-17.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-18 {
  width: 24px;
  height: 24px;
  color: #607D8B;
}

.emotion-19 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-20 {
  height: 0;
  overflow: hidden;
  -webkit-transition: height 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: height 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  visibility: hidden;
}

.emotion-21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.emotion-22 {
  width: 100%;
}

.emotion-23 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 16px;
  border-top: 1px solid #C2C7CF;
}

<div
    class="MuiContainer-root MuiContainer-maxWidthLg emotion-0"
  >
    <div
      class="MuiBox-root emotion-1"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 emotion-2"
        data-testid="privacy-policy-consent"
      >
        <h3
          class="MuiTypography-root MuiTypography-titleLargeSemiBold emotion-3"
        >
          นโยบายความเป็นส่วนตัว
        </h3>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth emotion-4"
        />
        <div
          class="MuiStack-root emotion-5"
        >
          <div
            class="MuiStack-root emotion-6"
          >
            <div
              class="emotion-7"
              data-testid="privacy-policy-content"
            >
              Consent Content
            </div>
          </div>
          <div
            class="MuiStack-root emotion-6"
            data-testid="privacy-policy-service"
          >
            <h3
              class="MuiTypography-root MuiTypography-titleLargeSemiBold emotion-9"
            >
              All Services
            </h3>
            <div
              class="MuiStack-root emotion-10"
            >
              <div
                class="MuiStack-root emotion-11"
                data-testid="privacy-policy-service-0"
              >
                <div
                  class="MuiBox-root emotion-12"
                >
                  <img
                    alt=""
                    class="emotion-13"
                    data-testid="privacy-policy-service-0-logo"
                    src="/service/logo/0"
                  />
                </div>
                <div
                  class="MuiStack-root emotion-14"
                >
                  <h3
                    class="MuiTypography-root MuiTypography-titleLargeSemiBold emotion-15"
                    data-testid="privacy-policy-service-0-title"
                  >
                    Service Title 0
                  </h3>
                  <svg
                    data-testid="privacy-policy-service-0-detail-icon"
                  />
                </div>
                <div
                  class="MuiStack-root emotion-16"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium emotion-17"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      class="emotion-18"
                    />
                    <span
                      class="MuiTouchRipple-root emotion-19"
                    />
                  </button>
                </div>
              </div>
              <div
                class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden emotion-20"
                style="min-height: 0px;"
              >
                <div
                  class="MuiCollapse-wrapper MuiCollapse-vertical emotion-21"
                >
                  <div
                    class="MuiCollapse-wrapperInner MuiCollapse-vertical emotion-22"
                  >
                    <div
                      class="MuiStack-root emotion-23"
                      data-testid="privacy-policy-service-0-content"
                    >
                      <div
                        class="emotion-7"
                      >
                        Consent Content 0
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
